x-pack/plugins/ml/public/application/util/dependency_cache.ts

```diff
@@ -220,3 +224,10 @@ export function getFileDataVisualizer() {
   }
   return cache.dataVisualizer;
 }
+
+export function getUsageCollection() {
+  if (cache.usageCollection === null) {
+    throw new Error("usageCollection hasn't been initialized");
+  }
+  return cache.usageCollection;
+}
```

Comment:
this function is not in use so we can delete it 
```suggestion
```

Reply:
I added knowing (hoping) it wouldn't be used, but just in case we wanted to add telemetry to something that doesn't have access to the kibana context.
but seeing as I added the `mlUsageCollection` service to the context which wraps the usage collection for ML's use, this function really shouldn't be used.
So yeah, i'll remove it.

Reply:
Changed in https://github.com/elastic/kibana/pull/107772/commits/b622cfd2a80be91a587e2db557bdc983c6f934a6